### PR TITLE
PP-8947 - Refactors FeeEntity

### PR DESCRIPTION
**Pull request by nimalank7**
Description:
- Uses Instant rather than `ZonedDateTime` in the same way that `ChargeEntity` does for createdDate
- Added a test for `FeeIncurredEventDetails`

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/events/model/charge/FeeIncurredEvent.java [hunk 1]
```diff
@@ -1,11 +1,32 @@
...
     }
+
+    public static FeeIncurredEvent from(ChargeEntity charge) {
+        Instant earliestInstant = charge.getFees()
+                .stream()
+                .map(FeeEntity::getCreatedDate)
+                .min(Instant::compareTo)
+                .orElse(null);
+
+        ZonedDateTime earliestCreatedDate = earliestInstant != null ? ZonedDateTime.ofInstant(earliestInstant, ZoneId.of("UTC")) : null;
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/connector/events/model/charge/FeeIncurredEvent.java]: If there are no fees, should we return an event at all? Surely there's never an occurrence where we're want to return the event with no created date, if there are no fees? Should it return null or throw an exception?

**Review by stephencdaly — COMMENTED** [review]
LGTM, just a question about what we should return for the event if there are no fees available

File: src/main/java/uk/gov/pay/connector/events/model/charge/FeeIncurredEvent.java (hunk 1 above)
Comment (nimalank7) [reply src/main/java/uk/gov/pay/connector/events/model/charge/FeeIncurredEvent.java]: Yep we've refactored it to throw an exception.

**Review by stephencdaly — APPROVED** [review]
LGTM

**Pull request merged**